include/nuttx/power/pm.h

```diff
@@ -85,6 +87,14 @@
  * own, custom idle loop to support board-specific IDLE time power management
  */
 
+#define PM_WAKELOCK_INITIALIZER(name, domain, state) {name, domain, state}
+
+#define PM_WAKELOCK_DECLARE(var, name, domain, state) \
+      struct pm_wakelock_s var = PM_WAKELOCK_INITIALIZER(name, domain, state)
+
+#define PM_WAKELOCK_DECLARE_STATIC(var, name, domain, state) \
```

Comment:
why do we need a special macro for static? I mean why `static PM_WAKELOCK_DECLARE(x, y, z, t);` can't be used?

Reply:
Since PM is optional feature, if we reuse PM_WAKELOCK_DECLARE macro, we have to:
```
#ifdef CONFIG_PM
static PM_WAKELOCK_DECLARE(x, y, z, t);
#endif
```
On the other hand, we can write more simple code with PM_WAKELOCK_DECLARE_STATIC:
```
PM_WAKELOCK_DECLARE_STATIC(x, y, z, t);
```